Add failing test to reproduce first-participant + initData problem

uekerman opened the pull request:
## Main changes of this PR

Adds the remaining failing test from #1307 
Needs breaking changes to be fixed, see also #1347


## Author's checklist

* [ ] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [ ] I ran `make format` to ensure everything is formatted correctly.
* [ ] I sticked to C++14 features.
* [ ] I sticked to CMake version 3.16.3.
* [ ] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

<!-- Tag people next to each point and add points for specific questions -->

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

<!-- add more questions/tasks if necessary -->

BenjaminRodenberg commented:
I just tried out merging #1350 @ [e8f738d](https://github.com/precice/precice/pull/1350/commits/e8f738d57e09d17dc9540ae77d2ad190f2df00f8) into this PR and the issue is resolved. I will not push the merge, because the changes are trivial and it just makes the history more complicated.

BenjaminRodenberg commented:
@uekerman what should we do with this PR? I think we can merge it into #1366. We can still keep the issue open, but it will be resolved in preCICE v3.0.0.

uekerman commented:
> What should we do with this PR? I think we can merge it into https://github.com/precice/precice/pull/1366. We can still keep the issue open, but it will be resolved in preCICE v3.0.0.

Yes, good point. I rebased on `develop-v3.0.0` and changed the target branch.
